src/main/java/org/openmaptiles/layers/Transportation.java

```diff
@@ -285,6 +287,22 @@ private static boolean isTrunkForZ5(String highway, List<RouteRelation> routeRel
       .anyMatch(Z5_TRUNK_BY_NETWORK::contains);
   }
 
+  /**
+   * Checks if a trunk segment is small enough to be processed at z5 so it can merge with surrounding motorways.
+   * 
+   * @param element the highway linestring element to check
+   * @return true if the trunk segment length is less than the threshold, false otherwise
+   */
+  private boolean isTrunkZ5MergeableLength(Tables.OsmHighwayLinestring element) {
+    try {
+      return element.source().length() < TRUNK_UPGRADE_LENGTH.apply(5).doubleValue();
```

Comment:
I'm not sure this does exactly what we want, `ZoomFunction.meterThresholds()` returns pixel lengths at a particular zoom, but when you're using it in process `length()` returns "world web mercator" length (where z0 tile goes from 0,0 to 1,1).

If my understanding is correct, then I think we want `TRUNK_UPGRADE_LENGTH = GeoUtils.metersToPixelAtEquator(0, 1_000) / 256` ? That would mean we upgrade any trunk to z4 motorway if it's effective length is <1000m

Reply:
Apart from the need to remove `TRUNK_GROUP_TEMP_KEY` before merging and some other minor stuff (see https://github.com/ZeLonewolf/planetiler-openmaptiles/pull/1 ) good for me. Thanks.

Reply:
Looks good to me, thanks for those fixes!

Reply:
It should be possible to do this without the finish handler/temp storage. I'll try to take a look tonight or tomorrow and see if I can figure out what's going on.

Reply:
Would something as simple as this work? https://github.com/ZeLonewolf/planetiler-openmaptiles/pull/2/changes (diff from main: https://github.com/openmaptiles/planetiler-openmaptiles/compare/main...msbarry:planetiler-openmaptiles:mikeb-omt-1767)

Here's Iowa at z5:

<img width="645" height="417" alt="image" src="https://github.com/user-attachments/assets/f1e2a5f0-d5bd-4ddd-b262-6955457ccc72" />

And I think this looks like what you want in Ontario?

<img width="858" height="633" alt="image" src="https://github.com/user-attachments/assets/18cd67d6-fd48-4300-a30f-6e4dd010a646" />

Reply:
It is a little weird because transcanada highway changes from motorway at z5 to trunk at z4, it might be missing a filter to only upgrade from trunk to motorway if segment length is less than some threshold

Reply:
OK latest commit should fix that